shell/platform/common/accessibility_bridge_unittests.cc

```diff
@@ -200,5 +265,37 @@ TEST(AccessibilityBridgeTest, canHandleSelectionChangeCorrectly) {
             ui::AXEventGenerator::Event::OTHER_ATTRIBUTE_CHANGED);
 }
 
+TEST(AccessibilityBridgeTest, doesNotAssignEditableRootToSelectableText) {
+  TestAccessibilityBridgeDelegate* delegate =
+      new TestAccessibilityBridgeDelegate();
+  std::unique_ptr<TestAccessibilityBridgeDelegate> ptr(delegate);
+  std::shared_ptr<AccessibilityBridge> bridge =
+      std::make_shared<AccessibilityBridge>(std::move(ptr));
+  FlutterSemanticsNode root;
+  root.id = 0;
+  root.flags = static_cast<FlutterSemanticsFlag>(
+      FlutterSemanticsFlag::kFlutterSemanticsFlagIsTextField |
+      FlutterSemanticsFlag::kFlutterSemanticsFlagIsReadOnly);
+  root.actions = static_cast<FlutterSemanticsAction>(0);
+  root.text_selection_base = -1;
+  root.text_selection_extent = -1;
+  root.label = "root";
+  root.hint = "";
+  root.value = "";
+  root.increased_value = "";
+  root.decreased_value = "";
+  root.child_count = 0;
+  root.custom_accessibility_actions_count = 0;
+  bridge->AddFlutterSemanticsNodeUpdate(&root);
+
+  bridge->CommitUpdates();
```

Comment:
Not introduced in this patch, and no need to change this in this patch, but I think we should shoot for emulating the framework and engine internal API here (a single update call that takes a list of nodes) as opposed to the embedding API (a series of update calls followed by a commit).